packages/lesmis/src/components/ProfileButton.js

```diff
@@ -26,14 +27,31 @@ const ProfileLinks = () => {
   );
 };
 
+const LoginLink = styled(MuiButton)`
+  color: white;
+  border-color: white;
+  padding: 0.5rem 2rem;
+  border-radius: 2.5rem;
+`;
+
 export const ProfileButton = () => {
+  const history = useHistory();
   const { data: user, isLoggedIn } = useUser();
   return user && isLoggedIn ? (
     <StyledProfileButton
       user={{ ...user, name: `${user.firstName} ${user.lastName}` }}
       MenuOptions={ProfileLinks}
     />
   ) : (
-    <LoginDialog buttonText="Log in" />
+    <LoginLink
+      variant="outlined"
+      component={RouterLink}
+      to={{
+        pathname: '/login',
+        state: { referer: history.location },
```

Comment:
It's a tidy little pattern. Out of interest, does it get represented in the URL at all, or is it just maintained internally?

Reply:
In react router, the state object is just maintained internally. Obviously, it could be done in the url as well but there didn't seem to be any advantage to going down that route ;)